packages/eslint-plugin/src/rules/contextual-decorator.ts

```diff
@@ -0,0 +1,123 @@
+import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+
+import {
+  getDecoratorName,
+  isAngularClassDecorator,
+  isAngularInnerClassDecorator,
+  isNotNullOrUndefined,
+  AngularClassDecoratorKeys,
+  ANGULAR_CLASS_DECORATOR_MAPPER,
+} from '../utils/utils';
+import { createESLintRule } from '../utils/create-eslint-rule';
+
+type Options = [];
+export type MessageIds = 'contextualDecorator';
+export const RULE_NAME = 'contextual-decorator';
+
+export default createESLintRule<Options, MessageIds>({
+  name: RULE_NAME,
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Ensures that classes use contextual decorators in its body.',
+      category: 'Best Practices',
+      recommended: 'error',
+    },
+    schema: [],
+    messages: {
+      contextualDecorator:
+        'Decorator out of context for "@{{classDecoratorName}}()"',
+    },
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      'MethodDefinition[kind="set"], MethodDefinition[kind="get"], MethodDefinition[kind="method"], ClassProperty, TSParameterProperty'(
+        node:
+          | TSESTree.MethodDefinition
+          | TSESTree.ClassProperty
+          | TSESTree.TSParameterProperty,
+      ) {
+        validateNode(context, node);
+      },
+    };
+  },
+});
+
+function validateNode(
+  context: TSESLint.RuleContext<MessageIds, []>,
+  node:
+    | TSESTree.MethodDefinition
+    | TSESTree.ClassProperty
+    | TSESTree.TSParameterProperty,
+): void {
+  if (!node.decorators || node.decorators.length === 0) {
+    return;
+  }
+
+  const classDeclaration = lookupTheClassDeclaration(node);
+
+  if (!classDeclaration) {
+    return;
+  }
+
+  const classDecoratorName = getClassDecoratorName(classDeclaration);
+
+  if (!classDecoratorName) {
+    return;
+  }
+
+  for (const decorator of node.decorators) {
+    validateDecorator(context, decorator, classDecoratorName);
+  }
+}
+
+function lookupTheClassDeclaration(
+  node: TSESTree.Node,
+): TSESTree.ClassDeclaration | null {
+  while (node.parent) {
+    if (node.type === 'ClassDeclaration') {
+      return node;
+    }
+
+    node = node.parent;
+  }
+
+  return null;
+}
+
+function getClassDecoratorName(
+  classDeclaration: TSESTree.ClassDeclaration,
+): AngularClassDecoratorKeys | undefined {
+  return (classDeclaration.decorators || [])
```

Comment:
```suggestion
  return (classDeclaration.decorators ?? [])
```

Reply:
In that particular case `||` will do the same thing as we don't care about empty string and `0` number :smile: (according to ECMA scec).

Reply:
Yeah, I prefer to be explicit but that's just a matter of preference and doesn't impact in anyway :)